tools/sesman/sesmanworker/engine.py

```diff
@@ -1760,3 +1774,198 @@ def get_target_str(self):
+    """
+
+    Filter field by field by applying one or several filters
+    on target string like account@domain@device:service:auth
+
+    For active the Target Accurate Filter mode, user MUST type on
+    selector research a string form beginning by '$' character like : 
+    "$keyword1:value1,$keyword2:value2,$keyword3:value3"
+
+    Each wanted filter must following the "$keyword:value" 
+    format for be applicable.
+
+    For example, the following target filter string will filter on
+    account, domain, device, service and authorization fields 
+    according to the passed values : 
+    "$account:my_account,$domain:my_domain,$device:my_device,$service:my_service,$auth:my_author"
+    
+    We can combine 2 filters or more like above but we 
+    can also execute only one filter on target.
+    The following target filter strings are accepted :
+    "$account:my_account"
+    "$domain:my_domain"
+    "$device:my_device"
+    "$service:my_service"
+    "$auth:my_author"
+
+    There is no specifc order for apply filters. 
+    The following target filter string is also valid :
+    "$device:my_device,$account:my_account,$domain:my_domain"
+
+    However, target filter string badly formatted like containing 
+    duplicated filter keywords, missing values and others aren't 
+    allowed and will be skipped and considered as no valid.
+    The following examples are forbidden :
+    "$account:my_account,$account:my_account"
+    "$domain:my_domain,"
+    "$my_device:my_device"
+    "$service:"
+    "$abcd:my_author"
+
+    """
+    
+    __slots__ = ("filter_pattern_dict")
+    
+    def __init__(self) :
+        self.filter_pattern_dict = {
+            "account" : None,
+            "domain" : None,
+            "device" : None,
+            "service" : None,
+            "auth" : None
+        }
+
+    def parse_target_filter_pattern(self, target_filter_pattern) :
+        FILTER_KV_SEPARATOR = ':'
+        FILTER_SEPARATOR = ','
+        filter_kv_list = target_filter_pattern.split(FILTER_SEPARATOR)
+
+        for filter_kv in filter_kv_list :
+            token_list = filter_kv.split(FILTER_KV_SEPARATOR, 1)
+
+            if len(token_list) != 2 :
+                raise ParsingError(filter_kv)
+
+            filter_keyword, filter_value = tuple(token_list)
+
+            if not filter_value :
+                raise ParsingError(filter_kv)
+
+            is_filter_keyword, filter_keyword_no_prefix = (
+                self._is_filter_keyword(filter_keyword))
+
+            if not is_filter_keyword :
+                raise FilterKeywordSyntaxError(filter_keyword)
+            if self.filter_pattern_dict[filter_keyword_no_prefix] is None :
+                self.filter_pattern_dict.update(
+                    {filter_keyword_no_prefix : filter_value})
+            else :
+                raise FilterValueDuplicateError(filter_keyword)
+
+    def is_filterable(self, target) :
+        if not self.filter_pattern_dict.values() :
```

Comment:
```py
if not self.filter_pattern_dict:
```

Reply:
If I replace it by your suggestion, it won't working because "filter_pattern_dict" MUST have keys specified in TargetAccurateFilter constructor

Reply:
Which is the case.

I realize that this condition is useless: it is always true, there are always values.

Reply:
The` raise NoFilterValueError(target)`  will be executed only if we forget to call parse_target_filter_pattern() before (assertion could be even better because it's the programmer fault if it's happens)

Reply:
```py
taf = TargetAccurateFilter()
taf.is_filterable('a@b@c')
```
Not call to `parse_target_filter_pattern()` and no error, this condition does nothing.

Reply:
Indeed, you're right... 

I changed a little the implementation about how to fill `self.filter_pattern_dict` : there is no `None` value anymore (`self.filter_pattern_dict = {}` at constructor now) which avoid some useless looping on `_is_filterable()` with `all()` built-in function